cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -103,7 +104,16 @@ public void getInfo(HttpRequest request, HttpResponder responder, @PathParam("na
     if (spec == null) {
       responder.sendStatus(HttpResponseStatus.NOT_FOUND);
     } else {
-      DatasetMeta info = new DatasetMeta(spec, implManager.getTypeInfo(spec.getType()), null);
+      // try finding type info in the dataset's namespace
```

Comment:
I think this comment is wrong. the method that the comment is for searches more than just the dataset's namespace.